src/js/letters/containers/AddressSection.jsx

```diff
@@ -43,38 +63,122 @@ export class AddressSection extends React.Component {
     }
   }
 
-  handleSave = () => {
-    this.setState({ isEditingAddress: false });
+  /**
+   * Runs all the valiations against the address passed as a prop for a given field.
+   *
+   * @param {String} fieldName             The name of the address field to validate.
+   *                                         Maps to the fieldValidations key.
+   * @param {Object} fullAddress            Contains the full mailing address.
+   * @param {Boolean} ignoreShouldValidate  Because we'll need to update the error
+   *                                         messages on multiple fields sometimes,
+   *                                         we need to run validations on all fields.
+   *                                         This ensures that we only run validation
+   *                                         if the field has been modified.
+   *                                         saveAddress will need to validate all
+   *                                         fields regardless of whether they've been
+   *                                         modified.
+   * @return {String|undefined}             If there's a validation error, return the
+   *                                         error message. If not, return undefined.
+   */
+  validateField = (fieldName, fullAddress, ignoreShouldValidate = false) => {
+    // Only validate the field if it's been modified
+    if (!this.state.shouldValidate[fieldName] || ignoreShouldValidate) {
+      return undefined;
+    }
+
+    const validations = fieldValidations[fieldName];
+    // If there is no validations array for that field, assume it has no validations
+    if (!validations) {
+      return undefined;
+    }
+
+    let errorMessage = false;
+    for (let i = 0; i < validations.length; i++) {
+      // this.props.value = address
+      errorMessage = validations[i](fullAddress[fieldName], fullAddress);
+      if (typeof errorMessage === 'string') {
```

Comment:
So will this just return the first errorMessage that is applicable to this field? But there could be multiple error messages if there are multiple validations?

Reply:
Yes, exactly. So this will return the first validation message, but if they have multiple validation errors, they'll have to be displayed / resolved one at a time. I thought it made more sense than something like "Please enter a zip code and please enter a valid zip code" or something, so we just show one message at a time.